### DISCUSSION TITLE
Gateway error (HTTP 502/504)

### DISCUSSION THREAD

**Discussion by baterflyrity** [deta/cloud-docs#323] [Help] [answered]
Hi there!

Using deta micro I periodically face HTTP GET errors: _502 Server Error: Bad Gateway_ and _504 Server Error: Gateway Time-out for url_. These errors are grouped by time interval - 5-10 minutes of errors then everything goes fine. Moreover cron runs my micro every 1 minute.

**Hence my question is: what couses these errors?** Is this cloud stability issues or  my provider's issue?

p.s.
Nothing is caught by visor.

**Comment by axmc1** [comment]
I had HTTP 502s too, and when I asked about it in the Discord, Mustafa connected me with Aavash and we had a quick talk and he said he'd look into it. Considering you're having similar issues, it shouldn't be an issue from our side. Could you confirm if the Micro(s) work fine even if facing those issues?

**Reply by baterflyrity** [reply]
Sure.

**Comment by abdelhai** [comment, marked as answer]
We did push and roll-back server updates the last few days, this could have been the cause. We will work on a zero-downtime deployment to avoid these issues. if you're still getting issues now, please let me know. Apologies for the inconvenience!

**Reply by baterflyrity** [reply]
Thanks for explanations.
